src/script/interpreter.cpp

```diff
@@ -1941,6 +1941,33 @@ uint256 GetSpentScriptsSHA256(const std::vector<CTxOut>& outputs_spent)
     return ss.GetSHA256();
 }
 
+/** Compute the vector where each element is SHA256 of scriptPubKeys spent by a tx. */
+std::vector<uint256> GetSpentScriptPubKeysSHA256(const std::vector<CTxOut>& outputs_spent)
+{
+    std::vector<uint256> spent_spk_single_hashes;
+    spent_spk_single_hashes.reserve(outputs_spent.size());
+    for (const auto& txout : outputs_spent) {
+        CHashWriter ss(SER_GETHASH, 0);
+        ss << txout.scriptPubKey;
+        spent_spk_single_hashes.push_back(ss.GetSHA256());
+    }
+    return spent_spk_single_hashes;
+}
+
+/** Compute the vector where each element is SHA256 of output scriptPubKey of a tx. */
+template <class T>
+std::vector<uint256> GetOutputScriptPubKeysSHA256(const T& txTo)
```

Comment:
Maybe take `const std::vector<CTxOut>& vout` parameter to avoid the template?

Reply:
Maybe it doesn't matter.

Reply:
This is consistent with `GetOutputsSHA256` is implemented.